sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -742,6 +742,7 @@ object FunctionRegistry {
     expression[SparkVersion]("version"),
     expression[TypeOf]("typeof"),
     expression[EqualNull]("equal_null"),
+    expression[SQLKeywords]("sql_keywords"),
```

Comment:
do other databases have such a function?

Reply:
It's spark-specific, other systems have other interfaces for end users. For example, using table `information_schema.keywords`

Reply:
I might add a little after I have asked ChatGPT for `the easiest way to list sql keywords from _ dynamically?`

- Postgres: SELECT word FROM pg_get_keywords() ;
- MySQL: SHOW Syntax or information_schema;
- Oracle: V$RESERVED_WORDS system, SELECT * FROM KEYWORDS;

Reply:
seems there is no standard way except for information schema. Let's not add this function.

Reply:
Although the function itself is not ANSI-standard, the behavior extending the standard to add useful functions complies with ANSI. Let's discuss whether to add it or not based on its usability.

- As we can see, most systems support such functionality with user-friendly SQL APIs by `command` or `functions` despite already having it in information_schema, which is not always for end-users.
- Comparing adding a new command with a new function, the latter is much more lightweight.
- The standard JDBC API only supports(by contract) list keywords but can not tell whether it is reserved or non-reserved. 
- the return type of sql_keywords is compliant with information_schema, it just acts as a `view` of information_schema.keywords.
- `sql_keywords` is a suitable name or not?

Reply:
then shall we add a TVF to return keywords? we can also include a boolean column `is_reserved`

Reply:
Make sense